x-pack/plugins/endpoint/server/services/resolver/search_handler.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { IScopedClusterClient } from 'kibana/server';
+import { CountResponse } from 'elasticsearch';
+import {
+  ResolverLegacyData,
+  ResolverElasticEndpointData,
+  ResolverData,
+  BaseResult,
+  ResolverResponse,
+} from '../../../common/types';
+import { EndpointAppContext, JSONish, Total } from '../../types';
+import { ResolverNode, ResolverDataHit } from './common';
+import { ResolverLegacyNode } from './legacy_node';
+import { ResolverElasticEndpointNode } from './es_endpoint_node';
+import { getPagination, PaginationInfo } from './query_builder';
+
+export interface ParentAndResolverData {
+  parentEntityID: string;
+  events: ResolverData[];
+}
+
+export abstract class BaseSearchHandler {
```

Comment:
maybe call this BaseResponseHandler or similar instead? It's not really responsible for executing searches which is confusing